### DISCUSSION TITLE
Filtering and saving chart screenshot using Rest API

### DISCUSSION THREAD

**Discussion by Palamariuk** [apache/superset#17806] [Q&A / Help]
Hi everyone! I'm trying to filter and save chart screenshot with Superset API. Actually there is no problem with the second step but I can't find any information about filtering. 
What is the best approach for this? Should I filter the chart itself or filter dashboard or maybe its dataset? 
Help me please with this problem

**Comment by nytai** [comment]
The screenshot api doesn't support additional filtering that what's been saved to the dashboard or chart. So you'll need to update the chart/dashboard with the filters you want before taking the screenshot

**Reply by Palamariuk** [reply]
Got it, thank you. But what api I should use to change chart filter? I can't find a good one in swagger
